### ISSUE TITLE
Instance Segmentation architectures implemented in R?

### ISSUE THREAD

**Issue by smorford** [rstudio/keras#767]
Hi all, 

This is a general question for the community here. I've been using U-Net successfully to build image segmentation models from aerial ecological imagery. I'd like to move towards doing instance segmentation (e.g. classify individual trees in aerial photos). I've been looking at some of the instance segmentation architectures that have been implemented in Keras (Mask R-CNN and MaskNet), but have not found clear examples for architecture implementation that are easily ported to R. Has anyone been working in this area and can provide some advice or pointers to useful resources? Thx.

**Comment by dfalbel** [comment]
We have 2 related blog posts here:

https://blogs.rstudio.com/tensorflow/posts/2018-11-05-naming-locating-objects/

https://blogs.rstudio.com/tensorflow/posts/2018-12-18-object-detection-concepts/

Hope it helps! Let us know if you need additional help :)

**Comment by smorford** [comment]
Thanks Daniel, 

Lots of great stuff to work through there. It's interesting how the complexity of the model code seems to inflate by an order of magnitude as you move to an instance-aware framework vs simple segmentation ala U-Net. Ultimately, I need to build a pixelwise mask to calculate areas, so these don't get me all the way to my goal, but a great place to start none the less.
